scripts/irbrc.rb

```diff
@@ -73,3 +38,40 @@
 rescue => exception
   puts "Error : 'load #{ENV["HOME"]}/.irbrc' : #{exception.message}"
 end
+
+# Turn on completion.
+begin
+  require "readline"
+
+  require "irb/completion" rescue nil
+
+  # Turn on history saving.
+  # require "irb/ext/save-history"
+  # IRB.conf[:HISTORY_FILE] = File.join(ENV["HOME"], ".irb-history")
+
+  # Use an alternate way to on history saving until save-history is fixed.
+  #
+  #   bug:   http://redmine.ruby-lang.org/issues/show/1556
+  #   patch: http://pastie.org/513500
+  #
+  # This technique was adopted from /etc/irbrc on OS X.
+  histfile = File.expand_path(".irb-history", ENV["HOME"])
+  histfile = IRB.conf[:HISTORY_FILE] || histfile
```

Comment:
There was no reference to `IRB.conf[:HISTORY_FILE]` in the original file. `IRB.conf[:HISTORY_FILE] = File.join(ENV["HOME"], ".irb-history")` would perpetuate the problem I'm attempting to fix. I think you are correct that `require "irb/ext/save-history"` is no longer required (for 1.9+). It MIGHT be necessary for 1.8?...

I am attempting to solve a very specific problem and was hoping to mitigate introducing other issues by minimizing the changes to this file. It hasn't been touched in 3 years, which would seem to indicate that while it may be simplified, there is nothing 'wrong' with most of what it's doing.

The lines regarding setting the history file that I changed:
Originally:
```
histfile = File.expand_path(".irb-history", ENV["HOME"])
```
to:
```
histfile = File.expand_path(".irb-history", ENV["HOME"])
histfile = IRB.conf[:HISTORY_FILE] || histfile
```
There are 100 ways to write this logic such that it still works. This is admittedly 'hack-ish', but I thought it better illustrated the change that I was making than other options.

My personal preference would really be something like this:
```
default_histfile = File.expand_path(".irb-history", ENV["HOME"])
histfile = IRB.conf[:HISTORY_FILE] || default_histfile
```
But technically, that's performing potentially unnecessary calculations, so if you care about that, something like this would better for performance:
(but 'performance' here is nano-seconds difference, especially since 99% of users will be evaluating the `File.expand_path` expression anyway)
```
histfile = IRB.conf[:HISTORY_FILE] || File.expand_path(".irb-history", ENV["HOME"])
```